ts/lib/src/index.ts

```diff
@@ -2,84 +2,163 @@ import * as secp from "@noble/secp256k1";
+  if (privateKey.length !== 32) {
+    throw new Error("Invalid private key length, expected 32 bytes");
+  }
+  return secp.getPublicKey(privateKey, true);
+}
+
+/**
+ * Gets the appropriate Crypto API to use.
+ * @returns The global Crypto API or the Node.js crypto module.
+ * If the global Crypto API is not available, it will import the Node.js crypto module.
+ */
 export async function getCrypto(): Promise<Crypto> {
   return typeof globalThis.crypto !== "undefined"
-  ? globalThis.crypto
-  : ((await import("node:crypto")).webcrypto as Crypto);
+    ? globalThis.crypto
+    : ((await import("node:crypto")).webcrypto as Crypto);
 }
 
-export function toHex(uint8: Uint8Array, withPrefix: boolean = false): string {
-  const hex = Array.from(uint8)
-  .map((byte) => byte.toString(16).padStart(2, "0"))
-  .join("");
+/**
+ * Converts a Uint8Array to a hex string.
+ * @param bytes The Uint8Array to convert.
+ * @param withPrefix If true, the hex string will be prefixed with "0x".
+ * @returns The hex string representation of the Uint8Array.
+ */
+export function toHexString(
+  bytes: Uint8Array,
+  withPrefix: boolean = false
+): string {
+  const hex = Array.from(bytes)
+    .map((byte) => byte.toString(16).padStart(2, "0"))
+    .join("");
   return withPrefix ? "0x" + hex : hex;
 }
 
-export function fromHex(hex: Uint8Array | string): Uint8Array {
-  return isBytes(hex)
-    ? Uint8Array.from(hex as any)
-    : fromStringHex(hex as string);
-}
-
-
-function isBytes(bytes: Uint8Array | string): boolean {
-  return (
-    bytes instanceof Uint8Array ||
-    (ArrayBuffer.isView(bytes) && bytes.constructor.name === "Uint8Array")
-  );
-}
-
-function fromStringHex(hex: string): Uint8Array {
+/**
+ * Converts a hex string to a Uint8Array.
+ * @param {string} hex The hex string to convert. It can optionally start with "0x" or "0X".
+ * @returns {Uint8Array} The Uint8Array representation of the hex string.
+ * @throws Error if the hex string has an odd length.
+ */
+export function fromHexString(hex: string): Uint8Array {
   hex = hex.startsWith("0x") || hex.startsWith("0X") ? hex.slice(2) : hex;
   if (hex.length % 2 !== 0) {
     throw new Error("Hex string must have an even length");
   }
   const bytes = new Uint8Array(hex.length / 2);
   for (let i = 0; i < hex.length; i += 2) {
-    bytes[i / 2] = parseInt(hex.slice(i, i + 2), 16);
+    try {
+      bytes[i / 2] = parseInt(hex.slice(i, i + 2), 16);
+    } catch {
+      throw new Error("Failed to parse hex string");
+    }
   }
   return bytes;
 }
 
+/**
+ * Ensures that the input is a Uint8Array.
+ * If the input is a hex string, it converts it to a Uint8Array.
+ * If the input is already a Uint8Array, it returns it as is.
+ * @param {Uint8Array | string} hex The input to ensure is a Uint8Array. It can be a hex string or a Uint8Array.
+ * @returns {Uint8Array} A Uint8Array representation of the input.
+ */
+export function ensureBytes(hex: Uint8Array | string): Uint8Array {
```

Comment:
Can this method throw exception? If not, I'd rather name it just simply `convertToBytes`, as `ensure` in the name means to me it's a read-only operation that throws if the object is not what's expected/ `convert` means that we're mutating something (even though we don't mutate input directly).